gui/control_main.py

```diff
@@ -1392,6 +1393,10 @@ def createSampleTab(self):
             lambda frame: self.updateCam(self.pixmap_item_HutchTop, frame)
         )
         hutchTopCamThread.start()
+        serverCheckThread = ServerCheckThread(
+            parent=self, delay=SERVER_CHECK_DELAY)
+        serverCheckThread.visit_dir_changed.connect(self.close)
```

Comment:
is `app.exit` (or similar) a better choice here? It does look like `def closeEvent` will exit, but directly exiting the app here may be more reliable?